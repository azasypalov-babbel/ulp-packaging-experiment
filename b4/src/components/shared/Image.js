import React from 'react';
import PropTypes from 'prop-types';

const Image = ({ imageId, imageUrl, noBubble, noFrame }) => {
  const frameClassName = noFrame ? '' : 'image-frame';
  const bubbleClassName = noBubble ? '' : 'image-frame-bubble';
  const className = `${frameClassName} ${bubbleClassName}`;

  return (
    <div className={className}>
      {imageId && <img alt="" src={imageUrl} />}
    </div>
  );
};

Image.propTypes = {
  imageId: PropTypes.string,
  imageUrl: PropTypes.string.isRequired,
  noBubble: PropTypes.bool,
  noFrame: PropTypes.bool
};

export default Image;
