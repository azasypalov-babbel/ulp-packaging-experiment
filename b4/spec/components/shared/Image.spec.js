import React from 'react';
import Image from '../../../src/components/shared/Image';
import { shallow } from 'enzyme';

const imageUrl = 'http://someurl.com';

const defaultProps = {
  imageId: '100afffb3',
  imageUrl: imageUrl
};

describe('<Image />', () => {
  let component;

  test('it renders', () => {
    component = shallow(
      <Image {...defaultProps} />
    );

    expect(component).toMatchSnapshot();
  });

  test('it renders without image-frame and image-frame-bubble', () => {
    component = shallow(
      <Image {...defaultProps} noFrame noBubble />
    );

    expect(component).toMatchSnapshot();
  });
});
