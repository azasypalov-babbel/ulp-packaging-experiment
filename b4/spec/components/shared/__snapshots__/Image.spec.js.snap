// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> it renders 1`] = `
<div
  className="image-frame image-frame-bubble"
>
  <img
    alt=""
    src="http://someurl.com"
  />
</div>
`;

exports[`<Image /> it renders without image-frame and image-frame-bubble 1`] = `
<div
  className=" "
>
  <img
    alt=""
    src="http://someurl.com"
  />
</div>
`;
